### PR TITLE
Streamer Auth Token

### LINKED ISSUE #450: API wrapper refactor: Try use streamer authorization for title/game updates
- request the requisite scope in `streamer_login`
- on auth fail with the bot's authorization, retry with the streamer's.
- If both fail: `bot.say` error message like right now.

### PR DESCRIPTION
**Pull request by TroyDota**
Pull request checklist:

- [ ] `CHANGELOG.md` was updated, if applicable
- [ ] Documentation in docs/ or install-docs/ was updated, if applicable

Fixes #450

### CODE REVIEW

**Comment by TroyDota** [comment]
Fixes #450

**Comment by RAnders00** [comment]
Looks good, but is it possible to try both bot and streamer auth? There's existing setups with the bot as channel editor, and that tends to be "lower friction" once it's set up since you can re-authenticate the bot yourself as the bot owner without having to bother the streamer.

**Comment by TroyDota** [comment]
I think we coo now?

**Comment by RAnders00** [comment]
now that I think about it, you could just inspect the token, since it knows what scopes are granted on it, and skip a streamer/bot token if the scope is not present at all

**Comment by TroyDota** [comment]
Well we don't know if the bot is s channel editor until we issue the command and twitch says no. Ill add the check for streamer token first see if scopes are present if not try update with bot. If this fails we return the error

**Comment by TroyDota** [comment]
@RAnders00 perhaps we catch error 401 and error 403? 
in pajbot/modules/basic/stream_update.py

File: pajbot/modules/basic/stream_update.py [hunk 1]
```diff
@@ -54,17 +54,19 @@ def generic_update(self, bot, source, message, field, api_fn):
...
-                return
-            else:
-                raise e
+        if "channel_editor" in bot.streamer_access_token_manager.token.scope:
+            api_fn(self.bot.streamer_user_id, message, authorization=bot.streamer_access_token_manager)
+        else:
+            try:
+                api_fn(self.bot.streamer_user_id, message, authorization=bot.bot_token_manager)
+            except HTTPError as e:
+                if e.response.status_code == 401:
```
Comment (TroyDota) [inline pajbot/modules/basic/stream_update.py:63]: Here.

Comment (TroyDota) [reply pajbot/modules/basic/stream_update.py:63]: 403 is when the bot is not a editor.

Comment (TroyDota) [reply pajbot/modules/basic/stream_update.py:63]: or add a response before we raise the error, or both.

**Comment by RAnders00** [comment]
https://github.com/pajbot/pajbot/blob/master/install-docs/README.md#further-steps This part also needs to be re-written to reflect these changes

**Comment by TroyDota** [comment]
Sorry I've been really busy with school rn I'll try get through all my prs
as soon as I've got a free :) hs is kinda aids with the amount of homework

**Comment by TroyDota** [comment]
I think thats good right?

**Comment by RAnders00** [comment]
Thanks for the contribution! This looks good now.

**Comment by TroyDota** [comment]
YEP

**Pull request merged**